src/draw/lv_draw_buf.h

```diff
@@ -152,6 +160,79 @@ void lv_draw_buf_clear(void * buf, uint32_t w, uint32_t h, lv_color_format_t col
 void lv_draw_buf_copy(void * dest_buf, uint32_t dest_w, uint32_t dest_h, const lv_area_t * dest_area_to_copy,
                       void * src_buf,  uint32_t src_w, uint32_t src_h, const lv_area_t * src_area_to_copy,
                       lv_color_format_t color_format);
+
+/**
+ * Note: Eventually, lv_draw_buf_malloc/free will be kept as private.
+ *       For now, we use `create` to distinguish with malloc.
+ *
+ * Create an draw buf by allocating struct for `lv_draw_buf_t` and allocating a buffer for it
+ * that meets specified requirements.
+ *
+ * @param w         the buffer width in pixels
+ * @param h
+ * @param cf        the color format for image
+ * @param buf_align the start address alignment for the buffer
+ * @param stride    the stride in bytes for image. Use 0 for automatic calculation based on
+ *                  w, cf, and global stride alignment configuration.
+ */
+lv_draw_buf_t * lv_draw_buf_create(uint32_t w, uint32_t h, lv_color_format_t cf,
+                                   uint32_t buf_align, uint32_t stride);
+
+/**
+ * Return a window to the main draw buffer.
+ * It always have same stride parameter as main buffer, thus can navigate to next line
+ * of window by adding stride to current pointer.
+ * @param buf   The main draw buffer
+ * @param area  The area of the window inside the draw buffer
+ * @return      A new draw buffer instance point to the window
+ */
+lv_draw_buf_t lv_draw_buf_window(lv_draw_buf_t * buf, const lv_area_t * area);
+
+/**
+ * Similar to `lv_draw_buf_create` but only allocating struct for `lv_draw_buf_t` and
+ * assemble it with provided buffer. Note, `stride` must not be zero and must match with `buf` stride.
+ *
+ * @param buf      the output of assembled draw buffer
+ * @param w        buffer width in pixels
+ * @param h        buffer height in pixels
+ * @param cf       color format for image
+ * @param flags    flags for image
+ * @param stride   stride in bytes for image. Must match real stride of `buf`.
+ * @param data     data buffer used for image
+ * @return         LV_RESULT_OK if succeeded, otherwise LV_RESULT_INVALID
+ */
+lv_result_t lv_draw_buf_assemble(lv_draw_buf_t * buf,
```

Comment:
```suggestion
lv_result_t lv_draw_buf_init(lv_draw_buf_t * buf,
```

We don't have "assemble" keyword yet.

Reply:
I have remove it. From current usage in decoder, I have seen no need for it. Let's add when it's really needed.